server/middlewares/csp.js

```diff
@@ -4,9 +4,13 @@ let trusted = [
   "'self'"
 ];
 
+const host = process.env.HOST || 'localhost';
+const port = process.env.PORT || process.env.SYNC_PORT || '3000';
+
 if (process.env.NODE_ENV !== 'production') {
   trusted = trusted.concat([
-    'ws://localhost:3000'
+    `http://${host}:${port}`,
```

Comment:
self is determined on the client. If the site is being served at `https://<worspace-name>-<username>.c9users.io` then self will point self to this origin.